File: clearhttp/src/main/java/com/csy6/clearhttp/MainActivity.java
package com.csy6.clearhttp;

import android.os.Bundle;
import android.view.View;
import android.widget.TextView;

import androidx.appcompat.app.AppCompatActivity;

import com.squareup.okhttp.Callback;
import com.squareup.okhttp.OkHttpClient;
import com.squareup.okhttp.Request;
import com.squareup.okhttp.Response;

import java.io.IOException;

/**
 * приложение демонстрирует уязвимость http траффика к сниффингу. Но такого вы почти не встретите в современных приложениях.
 * Для перехвата трафика ничего не требуется - просто настройте прокси на устройстве и смотрети траффик в бурпе
 */
public class MainActivity extends AppCompatActivity {
    public static final String URL = "http://api.bittrex.com/api/v1.1/public/getticker?market=BTC-LTC";
    OkHttpClient client;
    TextView tv_answer;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        tv_answer = findViewById(R.id.tv_answer);
    }

    public Request makeRequest(){
        Request request = new Request.Builder()
                .url(URL)
                .build();
        return request;
    }

    public void get(View view) {
        if(client == null) client = new OkHttpClient();
        client.newCall(makeRequest()).enqueue(new Callback() {
            @Override
            public void onFailure(Request request, IOException e) {
                tv_answer.append("\n"+e.getMessage());
            }

            @Override
            public void onResponse(Response response) throws IOException {
                tv_answer.append("\n"+response.body().string());
            }
        });
    }

}

File: okhttpcache/src/main/java/com/hackeru/okhttpcache/MainActivity.kt
package com.hackeru.okhttpcache

import android.os.Bundle
import android.util.Log
import android.view.View
import android.widget.EditText
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import okhttp3.*
import java.io.File
import java.io.IOException

class MainActivity : AppCompatActivity() {
    var client: OkHttpClient? = null
    lateinit var tv_log:TextView
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        tv_log = findViewById(R.id.tv_answer)
    }
    fun makeRequest(url:String): Request {
        return Request.Builder()
                .url(url)
                .build()
    }


    fun goToURl(view: View) {
        if (client == null) {
            client = OkHttpClient.Builder()
                    .cache(
                            Cache(directory = File(application.cacheDir, "interesting_data"), maxSize = 1024L * 1024 * 10)
                    ).build()
        }

        val url = findViewById<EditText>(R.id.ed_url).text.toString()
        Log.d("GO","go to url $url")
        client!!.newCall(makeRequest(url)).enqueue( object :Callback{
            override fun onFailure(call: Call, e: IOException) {
                Log.e("GO",e.toString())
                tv_log.append("${e.message}\n")
            }

            override fun onResponse(call: Call, response: Response) {
                Log.d("GO","OK ${response.code}")
                tv_log.append("answer: ${response.code}, ${response.headers["date"]}\n")
            }
        })
    }
}